test/e2e/reaper_test.go

```diff
@@ -2,46 +2,129 @@ package e2e
+
+	t.Log("check Reaper keyspace created")
+	f.CheckKeyspaceExists(t, ctx, "kind-k8ssandra-0", namespace, "test", "test-dc1-default-sts-0", "reaper_db")
+
+	testDeleteReaperManually(t, f, ctx, kcKey, dcKey, reaperKey)
+	testRemoveReaperFromK8ssandraCluster(t, f, ctx, kcKey, dcKey, reaperKey)
+
+	t.Log("deploying Reaper ingress routes in kind-k8ssandra-0")
+	f.DeployReaperIngresses(t, ctx, "kind-k8ssandra-0", 0, namespace, "test-dc1-reaper-service")
+	defer f.UndeployAllIngresses(t, "kind-k8ssandra-0", namespace)
+
+	testReaperApi(t, ctx, 0, "reaper_db")
+}
+
+func createMultiReaper(t *testing.T, ctx context.Context, namespace string, f *framework.E2eFramework) {
+
+	cqlSecretKey := types.NamespacedName{Namespace: namespace, Name: "reaper-cql-secret"}
+	jmxSecretKey := types.NamespacedName{Namespace: namespace, Name: "reaper-jmx-secret"}
+
+	checkSecretExists(t, f, ctx, framework.ClusterKey{K8sContext: "kind-k8ssandra-0", NamespacedName: cqlSecretKey})
+	checkSecretExists(t, f, ctx, framework.ClusterKey{K8sContext: "kind-k8ssandra-1", NamespacedName: cqlSecretKey})
+	checkSecretExists(t, f, ctx, framework.ClusterKey{K8sContext: "kind-k8ssandra-0", NamespacedName: jmxSecretKey})
+	checkSecretExists(t, f, ctx, framework.ClusterKey{K8sContext: "kind-k8ssandra-1", NamespacedName: jmxSecretKey})
+
+	kcKey := types.NamespacedName{Namespace: namespace, Name: "test"}
+	dc1Key := framework.ClusterKey{K8sContext: "kind-k8ssandra-0", NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc1"}}
+	dc2Key := framework.ClusterKey{K8sContext: "kind-k8ssandra-1", NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc2"}}
+	reaper1Key := framework.ClusterKey{K8sContext: "kind-k8ssandra-0", NamespacedName: types.NamespacedName{Namespace: namespace, Name: "test-dc1-reaper"}}
+	reaper2Key := framework.ClusterKey{K8sContext: "kind-k8ssandra-1", NamespacedName: types.NamespacedName{Namespace: namespace, Name: "test-dc2-reaper"}}
+
+	checkDatacenterReady(t, ctx, dc1Key, f)
+	checkDatacenterReady(t, ctx, dc2Key, f)
+
+	checkReaperReady(t, f, ctx, reaper1Key)
+	checkReaperReady(t, f, ctx, reaper2Key)
+
+	checkReaperK8cStatusReady(t, f, ctx, kcKey, dc1Key)
+	checkReaperK8cStatusReady(t, f, ctx, kcKey, dc2Key)
+
+	t.Log("check Reaper keyspace created")
+	f.CheckKeyspaceExists(t, ctx, "kind-k8ssandra-0", namespace, "test", "test-dc1-default-sts-0", "reaper_ks")
+	f.CheckKeyspaceExists(t, ctx, "kind-k8ssandra-1", namespace, "test", "test-dc2-default-sts-0", "reaper_ks")
+
+	t.Log("deploying Reaper ingress routes in kind-k8ssandra-0 and kind-k8ssandra-1")
+	f.DeployReaperIngresses(t, ctx, "kind-k8ssandra-0", 0, namespace, "test-dc1-reaper-service")
+	f.DeployReaperIngresses(t, ctx, "kind-k8ssandra-1", 1, namespace, "test-dc2-reaper-service")
+	defer f.UndeployAllIngresses(t, "kind-k8ssandra-0", namespace)
+	defer f.UndeployAllIngresses(t, "kind-k8ssandra-1", namespace)
+
+	testReaperApi(t, ctx, 0, "reaper_ks")
```

Comment:
Do you think it would be good to do the api verification in a subtest like you did for Stargate?